File: Data Structures & Algorithms/is-palindrome/submission-1.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string temp;
        for(int i=0; i<s.length(); i++) {
            if(isalnum(s[i])) {
                temp += s[i];
            }
        }
        for (auto& x : temp) {
            x = tolower(x);
        }
        int n = temp.length()-1;
        for(int i=0; i<temp.length()/2; i++) {
            if(temp[i] != temp[n]) {
                return false;
            }
            n--;
        }
        cout<<temp;
        return true;
    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp = head;
        int count = 0, k=0;

        while(temp != null) {
            count++;
            temp = temp.next;
        }
        if(count == n) return head.next;

        temp = head;
        while(temp != null) {
            if(k + 1 == count - n) {
                temp.next = temp.next.next;
            } 
            k++;
            temp = temp.next;
        }

        return head;
    }
}
